packages/components/src/Autocomplete/Menu.tsx

```diff
@@ -99,15 +101,45 @@ export function Menu({
           Math.min(options.length - 1, highlightedIndex + indexChange),
         );
       }
+      if (menuDiv.current) {
+        scrollMenuIfItemNotInView(menuDiv.current, "down");
+      }
     });
 
     useOnKeyDown("ArrowUp", (event: KeyboardEvent) => {
       const indexChange = arrowKeyPress(event, IndexChange.Previous);
       if (indexChange) {
         setHighlightedIndex(Math.max(0, highlightedIndex + indexChange));
       }
+      if (menuDiv.current) {
+        scrollMenuIfItemNotInView(menuDiv.current, "up");
+      }
     });
 
+    function scrollMenuIfItemNotInView(
+      menuDivElement: HTMLDivElement,
+      direction: "up" | "down",
+    ) {
+      const itemDiv = menuDivElement.querySelector(
+        `button.${classnames(styles.option)}:nth-child(${highlightedIndex +
+          1})`,
+      ) as HTMLButtonElement;
+      if (!itemDiv) {
+        return;
+      }
```

Comment:
Just personal preference for a guard statement.

```suggestion
      if (!itemDiv) return;
```

Reply:
Done!